pkg/event/target/mqtt.go

```diff
@@ -175,39 +182,35 @@ func NewMQTTTarget(id string, args MQTTArgs, doneCh <-chan struct{}, loggerOnce
 
 	client := mqtt.NewClient(options)
 
-	// The client should establish a first time connection.
-	// Connect() should be successful atleast once to publish events.
-	token := client.Connect()
-
 	target := &MQTTTarget{
 		id:         event.TargetID{ID: id, Name: "mqtt"},
 		args:       args,
 		client:     client,
 		loggerOnce: loggerOnce,
 	}
 
-	// Retries until the clientID gets registered.
-	retryRegister := func() {
-		// Repeat the pings until the client registers the clientId and receives a token.
+	go func(client mqtt.Client) {
 		for {
-			var terr error
+		retry:
 			select {
 			case <-doneCh:
 				return
 			default:
-				terr = token.Error()
-				if token.Wait() && terr == nil {
-					// Connected
-					return
+				token := client.Connect()
+				token.WaitTimeout(reconnectInterval * time.Second)
```

Comment:
What if the timeout happened sooner. The token.Error() would be nil in that case. token.Wait() is the only proof that the client has obtained its token successfully. Timeouts may hide the actual error in the background

Reply:
why would token.Error() be nil if token.WaitTimeout failed? @Praveenrajmani

Reply:
the point to note is we cannot have any code which indefinitely waits. So we should never use Wait() because that will simply lead to go-routine leaks.

Reply:
it can be `nil` or any connection timeout error. which btw is masking the actual error in the background.

Reply:
ok.. Having a reasonable timeout with an assumption that the call would succeed within that lapse would work here..

Reply:
Agreed :+1:

Reply:
@harshavardhana yes, we can't afford to wait indefinitely. But @Praveenrajmani is correct about how `token.Error` wouldn't be filled if the connection attempt has not returned with any success or failure yet.